pkg/jobconfig/files.go

```diff
@@ -499,3 +474,81 @@ func MakeRegexFilenameLabel(possibleRegex string) string {
 	}
 	return label
 }
+
+func isStale(job prowconfig.JobBase, newLabel ProwgenLabel) bool {
+	genLabel, generated := job.Labels[ProwJobLabelGenerated]
+	return generated && genLabel != string(newLabel)
+}
+
+func isGenerated(job prowconfig.JobBase) bool {
+	_, generated := job.Labels[ProwJobLabelGenerated]
+	return generated
+}
+
+func PruneStaleJobs(jobDir string, oldLabel, newLabel ProwgenLabel) error {
```

Comment:
Why are the labels parameterized here?

Reply:
I did not want to make pkg/jobconfig/files.go` coupled with prowgen `main.go` (more than it already is). The code in `main.go` determines how exactly are the jobs labelled, so I wanted to avoid the pkg to rely on that.

Reply:
But we won't ever prune with a different set of labels so the urge not to couple seems misplaced?

Reply:
I wouldn't say misplaced, but I hear what you say. How about I move this method to prowgen `main` and simplify it?